nat-lab/tests/telio.py

```diff
@@ -1048,6 +1050,76 @@ async def get_log(self) -> str:
         await process.execute()
         return process.get_stdout()
 
+    async def get_system_log(self) -> Optional[str]:
+        """
+        Get the system log on the target machine
+        Windows only for now
+        """
+        if self._connection.target_os == TargetOS.Windows:
+            output_notifier = OutputNotifier()
+            done_event = asyncio.Event()
+            done_event_str = "Get-EventLog - Done!"
+
+            async def on_stdout(stdout: str) -> None:
+                for line in stdout.splitlines():
+                    output_notifier.handle_output(line)
+
+            output_notifier.notify_output(
+                f"{done_event_str}",
+                done_event,
+            )
+
+            process = self._connection.create_process([
+                "powershell",
+            ])
+            async with process.run(
+                stdout_callback=on_stdout, stderr_callback=on_stdout
+            ):
+                await process.wait_stdin_ready()
+                await asyncio.sleep(1)
+                await process.write_stdin(
+                    "Get-EventLog -LogName Application -Newest 100 | format-table -wrap\n"
+                )
+                await process.write_stdin(
+                    "Get-EventLog -LogName System -Newest 100 | format-table -wrap\n"
+                )
+                await process.write_stdin(f"echo '{done_event_str}'\n")
+                await done_event.wait()
+                return process.get_stdout()
+        return None
+
+    async def clear_system_log(self) -> None:
+        """
+        Clear the system log on the target machine
+        Windows only for now
+        """
+        if self._connection.target_os == TargetOS.Windows:
+            output_notifier = OutputNotifier()
+            done_event = asyncio.Event()
+            done_event_str = "Clear-EventLog - Done!"
+
+            async def on_stdout(stdout: str) -> None:
+                for line in stdout.splitlines():
+                    output_notifier.handle_output(line)
+
+            output_notifier.notify_output(
+                f"{done_event_str}",
+                done_event,
+            )
+
+            process = self._connection.create_process([
+                "powershell",
+            ])
+            async with process.run(
+                stdout_callback=on_stdout, stderr_callback=on_stdout
+            ):
+                await process.wait_stdin_ready()
+                await asyncio.sleep(1)
```

Comment:
Left over or is it needed? If it is needed then why?

Reply:
`wait_stdin_ready` is more of a internal event, stating that `process` has been started and `stdin` should be ready. But that's not always the case. Like in this case, `powershell` process still needs some time to handle input unfortunately.

Reply:
Do we know exactly what is happening? Because it seems like some strange bug. If the process is not reading from stdin because it is busy with initialization the input data should just be waiting in a buffer and be consumed later. Where is it disappearing? Unless the powershell process is reading it and discarding during initialization but that would be very surprising behavior

Reply:
Now that I've tested it myself, sleep might be not needed. Something is wrong with following commands, when there is no event to be displayed. Will dig more into it.

Reply:
Rewrote it now, no sleeps are needed